### PR TITLE
Fix autoscroll

### PR DESCRIPTION
**Pull request by jrainville**
Fix autoscroll on load and on image load
Also hide the sent message on compact mode once the message is sent

### CODE REVIEW

File: ui/app/AppLayouts/Chat/ChatColumn/ChatMessages.qml [hunk 1]
```diff
@@ -43,18 +32,51 @@ ScrollView {
...
+                return
+            }
+            if (caller) {
+                if (caller !== chatLogView.itemAtIndex(chatLogView.count - 1)) {
+                    // If we have a caller, only accept its request if it's the last message
+                    return
+                }
+                // Add a small delay because images, even though they say they say they are loaed, they aren't shown yet
+                timer.setTimeout(function() {
+                    Qt.callLater(chatLogView.positionViewAtEnd)
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/ChatColumn/ChatMessages.qml:51]: Have you tried this with a really big image? Just curious if the `Image.Status` property waits for downloading to occur, and this additional `100ms` delay is simply there for the UI to update with the downloaded image.

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml [hunk 2]
```diff
@@ -31,19 +31,24 @@ Rectangle {
...
             anchors.top: (index == 0) ? parent.top: parent.children[index-1].bottom
             anchors.topMargin: imageChatBox.chatVerticalPadding
             sourceSize.width: imageChatBox.imageWidth
             source: modelData
             onStatusChanged: {
-                if (imageMessage.status == Image.Error) {
+                if (imageMessage.status === Image.Error) {
                     imageMessage.height = 0
+                    imageMessage.minHeight = 0
+                    imageMessage.source = ""
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:47]: On error, do you think it's a good idea if we replaced the image source with an error icon or something similar, to let the user know the image was not loaded?

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml [hunk 3]
```diff
@@ -31,19 +31,24 @@ Rectangle {
...
                     imageMessage.visible = false
                     imageChatBox.height = 0
                     imageChatBox.visible = false
-                } else if (imageMessage.status == Image.Ready) {
+                } else if (imageMessage.status === Image.Ready) {
                     messageItem.scrollToBottom(true, messageItem)
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:52]: Do we not need to update the `minHeight` property here? Could we possibly use the [`Image.paintedHeight`](https://doc.qt.io/qt-5/qml-qtquick-image.html#paintedHeight-prop) property?

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml [hunk 4]
```diff
@@ -31,19 +31,24 @@ Rectangle {
         }
 
         Image {
+            // This minimum height is for the initial load of the chat to work correctly
+            // Without this, the chat doesn't understand that the image wil lhave a height and doesn't scroll to the bottom
+            property int minHeight: 400
```
Comment (emizzle) [inline ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:36]: How does QML read this custom property in a way that affects scrolling? Is it possible to achieve a similar affect by setting the [`Image.height` property](https://doc.qt.io/qt-5/qml-qtquick-item.html#height-prop) instead?

Comment (jrainville) [reply ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:36]: QML doesn't understand `minHeight`, I just use it on line 14 if the image height is 0. I thought of trying `height` directly, but I don't know if doing `height: this.height || minHeight` works. I'll try

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml (hunk 3 above)
Comment (jrainville) [reply ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:52]: Nope, `minHeight` is only used if the height is 0

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml (hunk 2 above)
Comment (jrainville) [reply ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:47]: Good idea, I'll check if we have something like that in the design or icons

File: ui/app/AppLayouts/Chat/ChatColumn/ChatMessages.qml (hunk 1 above)
Comment (jrainville) [reply ui/app/AppLayouts/Chat/ChatColumn/ChatMessages.qml:51]: Just tested and it works

File: ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml (hunk 4 above)
Comment (jrainville) [reply ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/ImageMessage.qml:36]: Just tested, `height: this.height || minHeight` doesn't work

**Comment by jrainville** [comment]
@emizzle Thanks for the review. It made me improve it a bit. I also included ChatImage as well, since I forgot it existed.

Long story short: There is now a loading message during the image loading and then an error message if it failed.

I put that in a Component  that both MessageImage and Chat image use.

See the last 2 commits for that

**Pull request merged**